CMakeLists.txt

```diff
@@ -205,22 +206,23 @@ if(ENABLE_ZLIB_SUPPORT)
     endif(ZLIB_FOUND)
 endif(ENABLE_ZLIB_SUPPORT)
 
-add_subdirectory(include)
-
 add_subdirectory(3rdparty)
 include_directories(${thirdparty_inc})
 list(APPEND libs ${thirdparty_lib})
 add_definitions(${thirdparty_def})
 
 add_subdirectory(library)
+add_subdirectory(include)
 
 if(ENABLE_PROGRAMS)
     add_subdirectory(programs)
 endif()
 
-ADD_CUSTOM_TARGET(apidoc
-    COMMAND doxygen mbedtls.doxyfile
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/doxygen)
+if(ENABLE_DOCS)
```

Comment:
This doc enablement seems to be a change on its own. Could you move that to its own commit?

Reply:
Done. I've split it into 3 commits - I've gotten into bad habits of not properly partitioning commits

Reply:
I can see only 2 commits of yours: "Make api documentation target optional" and "Allow renaming of cmake targets". Otherwise, could you rebase on top of development? Commits are appearing in the commit list of this PR that are not related to it.

Reply:
I think I've sorted it now, I removed one commit since I think it's not a change that should be upstreamed in it's current form / wasn't related to the other commits